src/libs/libjpeg_turbo/lv_libjpeg_turbo.c

```diff
@@ -94,22 +106,15 @@ static lv_result_t decoder_info(lv_image_decoder_t * decoder, const void * src,
     if(src_type == LV_IMAGE_SRC_FILE) {
         const char * fn = src;
 
-        lv_fs_file_t f;
-        lv_fs_res_t res = lv_fs_open(&f, fn, LV_FS_MODE_RD);
-        if(res != LV_FS_RES_OK) {
-            LV_LOG_WARN("Can't open file: %s", fn);
+        uint32_t data_size;
+        uint8_t * data = alloc_file(fn, &data_size);
```

Comment:
**Performance degradation issue:** the whole file is read by `alloc_file()`, although this function first checks whether file is JPEG (reads first bytes), and only when JPEG signature is found, reads other parts of file and returns image meta data (width, height, etc...)

This `decoder_info()` function is called on source files to detect whether the file is JPEG, so it is called on PNG files (and other file types) too, and when the whole file is read and JPEG signature is not found, then reading the whole file was waste of resources (memory and FS calls).
Please keep the original logic, to read only the first X. bytes to check whether it is really a JPEG file.

Reply:
Well, if we need to read Exif information of a file, the original logic will open and close the file twice, I think it's not a good choice.

Reply:
I **do** think we should keep the original logic, because [lv_image_decoder.c -> lv_image_decoder_get_info()](https://github.com/lvgl/lvgl/blob/master/src/draw/lv_image_decoder.c#L93) function iterates through the decoders to find an applicable one (it tries decoders one-by-one), and if this libjpeg_turbo decoder is the first in the list and reads always the whole file (size could be >10 kB) and it can't find the JPEG signature (because the file we wanted to show is for example PNG), then it has read the whole file needlessly. LVGL FS cache is not a global cache (at the moment), so the cached buffer is released after file is closed, and the next decoder also reads part of the file.

A performance problem popped up also related to this `get_info()` callback, that it reads continously file content:
https://forum.lvgl.io/t/about-latest-multi-instance-updates/14258/2?u=pgnet

Reply:
Okay, I see. I will keep the original logic.